Source/ACE.Server/WorldObjects/Hotspot.cs

```diff
@@ -0,0 +1,154 @@
+    {
+        public Hotspot(Weenie weenie, ObjectGuid guid) : base(weenie, guid)
+        {
+            SetEphemeralValues();
+        }
+        public Hotspot(Biota biota) : base(biota)
+        {
+            SetEphemeralValues();
+        }
+        private void SetEphemeralValues()
+        {
+        }
+        public override void HandleActionOnCollideEnd(ObjectGuid playerId)
+        {
+            if (!playerId.IsPlayer()) return;
+            if (Players.Any(k => k == playerId))
+                Players.Remove(playerId);
+        }
+        private List<ObjectGuid> Players = new List<ObjectGuid>();
+        private ActionChain ActionLoop = null;
+        public override void HandleActionOnCollide(ObjectGuid playerId)
+        {
+            // to-do: pressure plates
+            if (!playerId.IsPlayer()) return;
+            if (!Players.Any(k => k == playerId))
+                Players.Add(playerId);
+            if (ActionLoop == null)
+            {
+                ActionLoop = NextActionLoop;
+                NextActionLoop.EnqueueChain();
+            }
+        }
+        private ActionChain NextActionLoop
+        {
+            get
+            {
+                ActionLoop = new ActionChain();
+                ActionLoop.AddDelaySeconds(CycleTimeNext);
+                ActionLoop.AddAction(this, () =>
+                {
+                    if (Players.Any())
+                    {
+                        Activate();
+                        NextActionLoop.EnqueueChain();
+                    }
+                    else
+                    {
+                        ActionLoop = null;
+                    }
+                });
+                return ActionLoop;
+            }
+        }
+        public string ActivationTalkString
+        {
+            get => GetProperty(PropertyString.ActivationTalk);
+            set { if (value == null) RemoveProperty(PropertyString.ActivationTalk); else SetProperty(PropertyString.ActivationTalk, value); }
+        }
+        private double CycleTimeNext
+        {
+            get
+            {
+                var variance = CycleTime * CycleTimeVariance;
+                var min = CycleTime - variance;
+                var max = CycleTime + variance;
+                return (double)Physics.Common.Random.RollDice((float)min, (float)max);
+            }
+        }
+        public double? CycleTime
+        {
+            get => GetProperty(PropertyFloat.HotspotCycleTime);
+            set { if (value == null) RemoveProperty(PropertyFloat.HotspotCycleTime); else SetProperty(PropertyFloat.HotspotCycleTime, (double)value); }
+        }
+        public double? CycleTimeVariance
+        {
+            get => GetProperty(PropertyFloat.HotspotCycleTimeVariance);
+            set { if (value == null) RemoveProperty(PropertyFloat.HotspotCycleTimeVariance); else SetProperty(PropertyFloat.HotspotCycleTimeVariance, (double)value); }
+        }
+        private int DamageNext
```

Comment:
This can use WorldObject.GetBaseDamage()

This is converting to an int very early in the damage transformation pipeline (with seemingly random floors/ceils)

In general, all of the damage calculations should be dealing with floats, and only at the very end should the final damage by Rounded.

Reply:
thanks for the heads up.  I'll assume you will delay until it's fixed.  Shouldn't be more than 8-15 hours from now.